File: android/app/src/main/java/voidawt/image/ImageProducer.java
package voidawt.image;

public interface ImageProducer {
    void addConsumer(ImageConsumer ic);

    boolean isConsumer(ImageConsumer ic);

    void removeConsumer(ImageConsumer ic);

    void startProduction(ImageConsumer ic);

    void requestTopDownLeftRightResend(ImageConsumer ic);
}

File: src/commonMain/kotlin/awt/image/ImageProducer.kt
package awt.image

expect interface ImageProducer {
    fun addConsumer(ic: ImageConsumer)
    fun isConsumer(ic: ImageConsumer): Boolean
    fun removeConsumer(ic: ImageConsumer)
    fun startProduction(ic: ImageConsumer)
    fun requestTopDownLeftRightResend(ic: ImageConsumer)
}